core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts

```diff
@@ -204,6 +226,44 @@ export class DotExperimentsConfigurationStore extends ComponentStore<DotExperime
         );
     });
 
+    // Goals
+    readonly setSelectedGoal = this.effect((selectedGoal$: Observable<{ goals: Goals }>) => {
+        return selectedGoal$.pipe(
+            tap(() =>
+                this.setSidebarStatus({
+                    status: Status.SAVING,
+                    experimentStep: ExperimentSteps.GOAL
+                })
+            ),
+            withLatestFrom(this.state$),
```

Comment:
Same as before, you should send this from the component.

Reply:
Done.